Frameworks/CoreGraphics/CGPath.mm

```diff
@@ -228,19 +233,20 @@ void CGPathAddLineToPoint(CGMutablePathRef path, const CGAffineTransform* m, flo
         return;
     }
 
-    if (m) {
-        CGPoint pt;
-
-        pt.x = x;
-        pt.y = y;
+    __preparePathForEditing(path);
 
+    CGPoint pt;
+    pt.x = x;
+    pt.y = y;
+    if (m) {
         pt = CGPointApplyAffineTransform(pt, *m);
-
-        x = pt.x;
-        y = pt.y;
     }
 
-    _CGPathAddElement(path, kCGPathElementAddLineToPoint, { x, y });
+    path->_geometrySink->BeginFigure(D2D1::Point2F(path->_currentPoint.x, path->_currentPoint.y), D2D1_FIGURE_BEGIN_FILLED);
+    path->_geometrySink->AddLine(D2D1::Point2F(pt.x, pt.y));
+    path->_geometrySink->EndFigure(D2D1_FIGURE_END_CLOSED);
```

Comment:
>EndFigure [](start = 25, length = 9)

Just nuke this function.  This is used elsewhere in UIKit, where this needs to be re-written using public apis (CGPathApply possibly) to get the exclusion path.